### PR TITLE
Add regexps

### PR DESCRIPTION
**Pull request by mrahul17**
@icyflame Please review

_Note to self_:
Add tests for regexps

### CODE REVIEW

File: regexps.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+    "re1": "([0-9]+) (day?|week?|month?|year?)(:?s)? ago",
+    "re2": "(?:last|previous) (Mon|Tues|Wednes|Thurs|Fri|Satur|Sun)(?:day)?",
```
Comment (icyflame) [inline regexps.json:3]: @light94 : I think we should use the standard three/four letter days. `mon, tue, Wed, Thurs, Fri, sat, sun`. (A couple of days need to change)

File: regexps.json [hunk 2]
```diff
@@ -0,0 +1,5 @@
...
+    "re3": "([0-9]+)(?:th)? (Jan(?:uary)?|Feb(?:ruary)?|March|April|May|June|July|August|Sept(?:ember)?|Oct(?:ober)?|Nov(?:ember)?|Dec(?:ember)?)"
```
Comment (icyflame) [inline regexps.json:4]: Same here. I think everything is there except `mar, Apr, Jun, Jul, aug`. That's about all. (I think a change here will also cause changes in the lists above.)

**Comment by icyflame** [comment]
@light94 Stellar patch. Just a couple changes, I have left line notes. :smile:

**Comment by mrahul17** [comment]
There is some issue with the regexps. I will reopen this when I have included the changes above.

**Comment by icyflame** [comment]
@light94 It could be that you haven't written them as `/[0-9]+/i`. I see that you have used the `new RegExp("", "")` function. Might want to see if we want to use that function or not. (I prefer using the `/regex/i` notation and not the function, if there isn't any advantage to that.)

**Comment by mrahul17** [comment]
The problem here is that a non capturing group inside a capturing group is actually being captured.
So  _Sat_ and _Satur_ need to be dealt with separately. One possible solution is to use multiple arrays an array of arrays for individual days. For example, index 6 of the days array could be in fact _['Sat','Satur']_ . I am looking for a better solution.

**Comment by mrahul17** [comment]
@icyflame , can you confirm my above statement. One way to test it is trying Satur and Sat, with non capturing group as (?:ur). Keeping it inside capturing group was capturing _ur_ as well, so multiple inputs in array were required.

**Comment by icyflame** [comment]
@light94 How about using the `|` operator? I am thinking something along the lines of `/(Mon|Tue|Wed|Thu|fri|sat|sun)(?:day|sday|rsday|nesday|urday)/i`

I know that this will match `Wedsday` and `Tuenesday` but if someone prefers to call Tuesday, Tuenesday, who are we to stop them, huh? 😄 

Anyway, I will check this and tell you. If you have the time, that's a solution worth trying.

**Comment by mrahul17** [comment]
Does this imply that we check only the first 3 / 4 letters?

**Comment by icyflame** [comment]
No, not really. I mean, we need to have either the three/four letter versions, or we need to have the full name of the day.

I am pretty sure that matching more won't qualify as a bug, as long as we don't match Tuesday to a wrong day, and that won't happen here. So, I think this is good enough.

**Comment by icyflame** [comment]
@light94 Okay, I have finally come to a decision on how to do this. Let the regex be the same, let it capture the non-capturing groups as well, if required.

But before matching with our array, strip it to only the first three characters. That way we can maintain our match criterion, as well as, remove the present redundancy.

**Pull request closed without merging**